dask/dataframe/io/tests/test_parquet.py

```diff
@@ -111,3 +111,17 @@ def test_optimize(fn, c):
     dsk = x._optimize(x.dask, x._keys())
     assert len(dsk) == x.npartitions
     assert all(v[4] == c for v in dsk.values())
+
+
+def test_categorical():
+    with tmpdir() as tmp:
+        df = pd.DataFrame({'x': ['a', 'b', 'c'] * 100},
+                          dtype='category')
+        ddf = dd.from_pandas(df, npartitions=3)
+        to_parquet(tmp, ddf)
+
+        ddf2 = read_parquet(tmp, categories=['x'])
+
+        assert ddf2.x.cat.categories.tolist() == ['a', 'b', 'c']
+        out = ddf2.loc[:1000].compute()
+        assert all(out.x == df.x)
```

Comment:
The indexes can be subtly different - in particular, if the inputed index had no name, it comes back named "index"; if there is no index at all (by choice, of because the index was default sequential range), then the numbering will start from 0 in each division. That could in theory be fixed, since we know the row count in each division.
Fixed for now.